examples/puppetmaster/test.sh

```diff
@@ -124,6 +127,25 @@ install_required_module_dependency() {
   run_in_puppet puppet module install puppetlabs-registry
 }
 
+get_docker_gateway_ip() {
+  # Get the IP address of the Docker compose network's gateway.
+  DOCKER_GATEWAY_IP="$(docker inspect $(docker-compose ps -q puppet)| \
+    jq .[0].NetworkSettings.Networks[].Gateway | tr -d '"')"
+}
+
+add_puppetmaster_etc_hosts() {
+  # Make an /etc/hosts entry in the Puppet server container for the DNS name
+  # 'conjur.cyberark.com'. This is used by the Puppet server to access the
+  echo "Adding /etc/hosts entry in puppet server: \"$DOCKER_GATEWAY_IP conjur.cyberark.com\""
+  # Make a temporary copy of /etc/hosts to modify since `sed` is not able to
+  # directly modify /etc/hosts when it is run via 'docker exec'.
+  run_in_puppet bash -c \
+    "cp /etc/hosts /tmp/hosts; \
+    /bin/sed -i $'/\tconjur.cyberark.com$/d' /tmp/hosts; \
+    echo $'$DOCKER_GATEWAY_IP\tconjur.cyberark.com' >> /tmp/hosts; \
```

Comment:
What's the `$` here for?

Reply:
The $'...' pattern is ANSI-C quoting that I needed to include a literal tab (\t) character in the sed pattern (https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html#ANSI_002dC-Quoting).

I'll change this to be clearer by using the ANSI-C quoting only around the tab character:
```
     echo $DOCKER_GATEWAY_IP$'\t'conjur.cyberark.com >> /tmp/hosts; \
```
and do the same in the line above this one.

Reply:
TIL!